Description: adb: backward compatibility with openssl < 1.1
  A change introduced previously (adb_libssl_11.diff) made
  changes that would not work with openssl < 1.1, so add the
  one function we use for that case.
Origin: other, https://wiki.openssl.org/index.php/1.1_API_Changes
Last-Update: 2016-11-23
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/adb/adb_auth_host.cpp
+++ b/adb/adb_auth_host.cpp
@@ -56,6 +56,19 @@
 #include <openssl/base64.h>
 #endif
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+void RSA_get0_key(const RSA *r,
+                 const BIGNUM **n, const BIGNUM **e, const BIGNUM **d)
+{
+   if (n != NULL)
+       *n = r->n;
+   if (e != NULL)
+       *e = r->e;
+   if (d != NULL)
+       *d = r->d;
+}
+#endif
+
 #define ANDROID_PATH   ".android"
 #define ADB_KEY_FILE   "adbkey"
 
